File: SceytChatUiKit/src/main/java/com/sceyt/sceytchatuikit/media/audio/AudioPlayer.java
package com.sceyt.sceytchatuikit.media.audio;

public interface AudioPlayer {
    // Constructor accepts created File or file path

    boolean initialize();

    void play();

    void pause();

    void stop();

    long getPlaybackPosition();

    long getAudioDuration();

    void seekToPosition(long position);

    void togglePlayPause();

    void setPlaybackSpeed(float speed);

    String getFilePath();

    boolean isPlaying();

    void addEventListener(AudioPlayerHelper.OnAudioPlayer event, String tag);
}

File: SceytChatUiKit/src/main/java/com/sceyt/chatuikit/media/audio/AudioPlayer.kt
package com.sceyt.chatuikit.media.audio

import com.sceyt.chatuikit.persistence.logicimpl.message.MessageTid

interface AudioPlayer {
    // Constructor accepts created File or file path
    fun initialize(): Boolean
    fun play()
    fun pause()
    fun stop(savedState: AudioPlaybackState? = null)
    fun seekToPosition(position: Long)
    fun togglePlayPause()
    fun setPlaybackSpeed(speed: Float)
    fun getPlaybackSpeed(): Float
    fun getPlaybackPosition(): Long
    fun getAudioDuration(): Long
    fun getFilePath(): String?
    fun getMessageTid(): MessageTid
    fun isPlaying(): Boolean
}
